doc/developer/scan.txt

```diff
@@ -132,22 +63,266 @@ files, and section of the code they deal with, are :
+However, in some cases, there are outputs for which there is only really a
+need to store the most recent N values, not all of them.
+
+For instance, if a Scan node has a sitsot output (last computed value is
+fed back as an input at the next iteration) and only the last timestep of
+that output is ever used in the outer function, the ScanSaveMem optimization
+could determine that there is no need to store all computed timesteps for
+that sitsot output. Only the most recently computed timestep ever needs to
+be kept in memory.
+
+
+ScanMerge
+---------
+
+This optimization attempts to fuse distinct Scan ops into a single Scan op
+that performs all the computation. The main advantage of merging Scan ops
+together comes from the possibility of both original ops having some
+computation in common. In such a setting, this computation ends up being done
+twice. The fused Scan op, however, would only need to do it once and could
+therefore be more computationally efficient. Also, since every Scan node
+involves a certain overhead, at runtime, reducing the number of Scan nodes in
+the graph can improve performance.
+
+
+scan_merge_inouts
+-----------------
+
+This optimization attempts to merge a Scan op's identical outer inputs as well
+as merge its identical outer outputs (outputs that perform the same
+computation on the same inputs). This can reduce the amount of computation as
+well as result in a simpler graph for both the inner function and the outer
+function.
+
+
+Helper classes and functions
+============================
+
+Because of the complexity involved in dealing with Scan, a large number of
+helper classes and functions have been developped over time to implement
+operations commonly needed when dealing with the Scan op. The Scan op
+itself defines a large number of them and others can be found in the file
+`scan_utils.py`. This sections aims to point out the most useful ones and
+organize them according to their uses.
+
+
+Accessing/manipulating Scan's inputs and outputs by type
+--------------------------------------------------------
+
+Declared in `scan_utils.py`, the class `scan_args` handles the parsing
+of the inputs and outputs (both inner and outer) to a format that is easier to
+analyse and manipulate. Without this class, analysing scan's inputs and
+outputs often required convoluted logic which make for code that is hard to
+read and to maintain. Because of this, you should favor using `scan_args` when
+it is practical and appropriate to do so.
+
+The Scan op also defines a few helper functions for this purpose, such as
+`inner_nitsot_outs()` or `mitmot_out_taps()`, but they are often poorly
+documented and easy to misuse. These should be used with great care.
+
+
+Navigating between outer inputs/outputs and inner inputs/outputs
+----------------------------------------------------------------
+
+Navigation between these four sets of variables can be done in two ways,
```

Comment:
Maybe adding a small sentence describing what you mean by navigating between variables would be a good idea. Is it a copy? A casting? A change of scope?